src/views/pools/PoolsOverview.tsx

```diff
@@ -1,32 +1,178 @@
   }
   const clickStakeHandler = (asset: string) => {
     history.push(stakeRoutes.stake.path({ asset }))
   }
+  const btnCol = {
+    key: 'swap',
+    title: (
+      <ActionColumn>
+        <Button onClick={() => {}} typevalue="outline">
+          <SyncOutlined />
+          refresh
+        </Button>
+      </ActionColumn>
+    ),
+    render: (text: string, record: PoolRowType) => {
+      const {
+        pool: { asset, target }
+      } = record
+
+      return (
+        <TableAction>
+          <Button round onClick={() => clickStakeHandler(target)} typevalue="outline">
+            <PlusOutlined />
+            liquidity
+          </Button>
+          <Button round onClick={() => clickSwapHandler({ source: asset, target: target })}>
+            <SwapOutlined />
+            swap
+          </Button>
+        </TableAction>
+      )
+    }
+  }
+  const columns: ColumnsType<PoolRowType> = [
+    {
+      key: 'pool',
+      title: 'pool',
+      dataIndex: 'pool',
+      render: ({ target }: { asset: string; target: string }) => <Coin type="rune" over={target} />
+    },
+    {
+      key: 'asset',
+      title: 'asset',
+      dataIndex: 'pool',
+      render: ({ target }: { target: string }) => <p>{target}</p>,
+      sorter: (a: PoolRowType, b: PoolRowType) => a.pool.target.localeCompare(b.pool.target),
+      sortDirections: ['descend', 'ascend']
+    },
+    {
+      key: 'poolprice',
+      title: 'pool price',
+      dataIndex: 'poolPrice',
+      sorter: (a: PoolRowType, b: PoolRowType) => a.raw.poolPrice.minus(b.raw.poolPrice).toNumber(),
+      sortDirections: ['descend', 'ascend'],
+      defaultSortOrder: 'descend'
+    },
+    {
+      key: 'depth',
+      title: 'depth',
+      dataIndex: 'depth',
+      sorter: (a: PoolRowType, b: PoolRowType) => a.raw.depth.minus(b.raw.depth).toNumber(),
+      sortDirections: ['descend', 'ascend']
+    },
+    {
+      key: 'vol',
+      title: '24h vol',
+      dataIndex: 'volume',
+      sorter: (a: PoolRowType, b: PoolRowType) => a.raw.volume.minus(b.raw.volume).toNumber(),
+      sortDirections: ['descend', 'ascend']
+    },
+    {
+      key: 'transaction',
+      title: 'avg. transaction',
+      dataIndex: 'transaction',
+      sorter: (a: PoolRowType, b: PoolRowType) => a.raw.transaction.minus(b.raw.transaction).toNumber(),
+      sortDirections: ['descend', 'ascend']
+    },
+    {
+      key: 'slip',
+      title: 'avg. slip',
+      dataIndex: 'slip',
+      render: (slip: string) => <Trend amount={bn(slip)} />,
+      sorter: (a: PoolRowType, b: PoolRowType) => a.raw.slip.minus(b.raw.slip).toNumber(),
+      sortDirections: ['descend', 'ascend']
+    },
+    {
+      key: 'trade',
+      title: 'no. of trades',
+      dataIndex: 'trade',
+      sorter: (a: PoolRowType, b: PoolRowType) => a.raw.trade.minus(b.raw.trade).toNumber(),
+      sortDirections: ['descend', 'ascend']
+    },
+    btnCol
+  ]
+
+  const getPoolViewData = (pools: PoolsState): PoolRowType[] => {
```

Comment:
Extract this function into a helper module and write a test for it. If you like to do that with another PR, create an issue for this, pls.

Reply:
Issue https://github.com/thorchain/asgardex-electron/issues/158